packages/core/src/auth/tokens/token-scope.ts

```diff
@@ -1,26 +1,66 @@
 export enum CacheRole {
   None = 'none',
   ReadWrite = 'readwrite',
+  ReadOnly = 'readonly',
 }
 
+export class All {}
+export class CacheName {
+  name: string;
+  constructor(name: string) {
+    this.name = name;
+  }
+}
+export type CacheResource = All | CacheName;
+
 export class CachePermission {
   cacheRole: CacheRole;
+  cache: CacheResource;
 
-  constructor(cacheRole: CacheRole) {
+  constructor(cacheRole: CacheRole, cache?: CacheResource) {
     this.cacheRole = cacheRole;
+    if (cache) {
```

Comment:
nit, I think you could remove the `?` from the argument type and just put a default value directly in the signature?

Reply:
if you end up making an Options interface for the Topic one as per my comment below, you will want to do it here too for consistency. In which case you won't be able to set a default value in the function signature and you can ignore my comment about that above :)

Reply:
Is one preferred over the other w.r.t. default values vs interface? I don't have strong opinions on Javascript so I can follow whatever the standards are.

Reply:
i would go with options interface, it is definitely the most common pattern when you have multiple optional arguments like you do in the topic one.